sonar-slang-plugin/src/main/java/com/sonarsource/slang/kotlin/KotlinTreeVisitor.java

```diff
@@ -171,7 +172,7 @@ private Tree createFunctionDeclarationTree(TreeMetaData metaData, KtFunction fun
     PsiElement nameIdentifier = functionElement.getNameIdentifier();
     Tree returnType = null;
     IdentifierTree identifierTree = null;
-    List<Tree> parametersList = list(functionElement.getValueParameters().stream());
+    List<ParameterTree> parametersList = (List<ParameterTree>)(List<?>)list(functionElement.getValueParameters().stream());
```

Comment:
This should not work as a `ParameterTreeImpl` is never creted.
You can revert this change for now, and do another PR just related to kotlin AST convertion